stubs/keyboard/keyboard/__init__.pyi

```diff
@@ -0,0 +1,169 @@
+from collections import Counter, defaultdict, deque
+from collections.abc import Callable, Generator, Iterable, Sequence
+from queue import Queue
+from threading import Event as _UninterruptibleEvent
+from typing import Optional
+from typing_extensions import Literal, TypeAlias
+
+from ._canonical_names import all_modifiers as all_modifiers, sided_modifiers as sided_modifiers
+from ._generic import GenericListener as _GenericListener
+from ._keyboard_event import KEY_DOWN as KEY_DOWN, KEY_UP as KEY_UP, KeyboardEvent as KeyboardEvent
+
+_Key: TypeAlias = int | str
+_ScanCodeList: TypeAlias = list[int] | tuple[int, ...]
+_ParseableHotkey: TypeAlias = _Key | list[int | _ScanCodeList] | tuple[int | _ScanCodeList, ...]
+_Callback: TypeAlias = Callable[[KeyboardEvent], Optional[bool]] | Callable[[], Optional[bool]]
+# Can't use ParamSpecArgs on `args`, only on `*args`
+# _P = ParamSpec("_P")
+_P: TypeAlias = tuple[object, ...]
+
+version: str
+
+class _Event(_UninterruptibleEvent):
+    def wait(self) -> None: ...  # type: ignore[override]  # Actual implementation
+
+def is_modifier(key: _Key | None) -> bool: ...
+
+class _KeyboardListener(_GenericListener):
+    transition_table: dict[
+        tuple[Literal["free"], Literal["up"], Literal["modifier"]]
```

Comment:
This is a little excessive. How do we help users by duplicating this whole table at the type level?